support/metrics/metricsTracker.go

```diff
@@ -246,34 +295,45 @@ func (mt *MetricsTracker) SendUpdateEvent(now time.Time, success bool, millisFor
 
 // SendDeleteEvent sends the delete Amplitude event.
 func (mt *MetricsTracker) SendDeleteEvent(exit bool) error {
-	commonProps := mt.props
-	commonProps.SecondsSinceStart = time.Now().Sub(mt.botStartTime).Seconds()
 	deleteProps := deleteProps{
-		commonProps: commonProps,
-		Exit:        exit,
-		StackTrace:  string(debug.Stack()),
+		Exit:       exit,
+		StackTrace: string(debug.Stack()),
 	}
 
-	return mt.sendEvent(deleteEventName, deleteProps)
+	return mt.SendEvent(deleteEventName, deleteProps, time.Now())
 }
 
-func (mt *MetricsTracker) sendEvent(eventType string, eventProps interface{}) error {
+// SendEvent sends an event with its type and properties to Amplitude.
+func (mt *MetricsTracker) SendEvent(eventType string, eventPropsInterface interface{}, now time.Time) error {
 	if mt.apiKey == "" || mt.userID == "-1" || mt.isDisabled {
 		log.Printf("metric - not sending event metric of type '%s' because metrics are disabled", eventType)
 		return nil
 	}
 
+	trackerProps := mt.props
+	trackerProps[secondsSinceStartKey] = now.Sub(mt.botStartTime).Seconds()
+
+	eventProps, e := utils.ToMapStringInterface(eventPropsInterface)
+	if e != nil {
+		return fmt.Errorf("could not convert event props to map: %s", e)
+	}
+
+	mergedProps, e := utils.MergeMaps(trackerProps, eventProps)
+	if e != nil {
+		return fmt.Errorf("could not merge event properties: %s", e)
+	}
+
 	// session_id is the start time of the session in milliseconds since epoch (Unix Timestamp),
 	// necessary to associate events with a particular system (taken from amplitude docs)
 	eventW := eventWrapper{
-		ApiKey: mt.apiKey,
+		APIKey: mt.apiKey,
 		Events: []event{{
 			UserID:    mt.userID,
 			SessionID: mt.botStartTime.Unix() * 1000, // convert to millis based on docs
 			DeviceID:  mt.deviceID,
 			EventType: eventType,
-			Props:     eventProps,
-			Version:   mt.props.CliVersion,
+			Props:     mergedProps,
+			Version:   mt.cliVersion,
```

Comment:
cliVersion is not added on the gui factory method so this will produce an empty string